source-control/readme.md

```diff
@@ -32,11 +32,63 @@ The team should at least be doing the following:
 
 ## Commit Best Practices
 
-TODO: Add more content and details here!
+* Make small commits. This makes changes easier to review, and if we need to revert a commit, we loose less work.
+* Commit complete and well tested code. Never commit incomplete code, get in the habit of testing your code before commiting.
+* Don't mix whitespace changes with functional code changes. It is hard to determine if the line has a functional change or only removes a whitespace, so funcitonal changes may go unnoticed.
+* Write good commit messages.
 
-* Link work items
-* How often to commit
-* When to push
+A good commit message should answer these questions:
+
+* Why is it neccessary? It may fix a bug, add a feature, improve performance, or just be a change for the sake of correctness
+* How does it adress the issue? For short, obviuos changes, this can be omitted
+* What effects does this change have? In addition to the obvious ones, this may include benchmarks, side effects etc.
+* What limitations does the current code have?
+
+Consider this when writing your commit message:
+
+* Don't assume that the code is self-evident/self-documenting
+* If it seems difficult to summarize your commit, it may be because it includes more than one logical change or bug fix. If so, it is better to split it into separate commits with `git add -p`
+* Don't assume the reviewer understands the original problem.
+  It should be possible to review a change request without reading the contents of the original bug/task.
+
+Good message structure:
+
+* Separate subject from body with a blank line
+* Limit the subject line to 50 characters
+* Capitalize the subject line
+* Do not end the subject line with a period
+* Use the imperative mood in the subject line (*Fix typo in log* vs. *Fixed typo in log* or *Misc fixes in log code*)
```

Comment:
Just curious: Where does this come from? Perhaps, I've being doing it wrong (or it's just matter of taste), but I tend to use the 3rd person form "Fixes typo in log" as that's what the change will do.

To be clear, I'm not requesting a change here.

Reply:
From here https://chris.beams.io/posts/git-commit/ - using imperative mood is becoming a bit of a de facto standard I think... This particular path was just moved from the git doc.  I personally am not a big fan of the long messages - but left it there as the Chris Beams suggestions seem to be what most people recommend